### ISSUE TITLE
Squiz.Arrays.ArrayDeclaration sniff gives incorrect NoComma error for multiline string values

### ISSUE THREAD

**Issue by JDGrimes** [squizlabs/PHP_CodeSniffer#584]
Consider this code (I didn't write it):

```php
$var = array(
	'tab_template'      => '
		<li>%s</li>',
	'panel_template'        => '
		<div id="%s">
		    %s
		</div>',
);
```

You will get two `Squiz.Arrays.ArrayDeclaration.NoComma` errors, even though both array values are followed by commas. 

This change fixes it for me, but I don't know if it breaks anything else:

```diff
Index: CodeSniffer/Standards/Squiz/Sniffs/Arrays/ArrayDeclarationSniff.php
<+>UTF-8
===================================================================
--- CodeSniffer/Standards/Squiz/Sniffs/Arrays/ArrayDeclarationSniff.php	(revision 8b75b9c1d842642ecf8997bc3691f0077a5f2546)
+++ CodeSniffer/Standards/Squiz/Sniffs/Arrays/ArrayDeclarationSniff.php	(revision )
@@ -837,7 +837,7 @@
                 }
             }//end for
 
-            if ($nextComma === false || ($tokens[$nextComma]['line'] !== $valueLine)) {
+            if ($nextComma === false) {
                 $error = 'Each line in an array declaration must end in a comma';
                 $fix   = $phpcsFile->addFixableError($error, $index['value'], 'NoComma');
 
```

Related: #582.

**Comment by aik099** [comment]
That's tricky, because it's a multi-line string, that is tokenized as single token and only line number it has is when it starts.

**Comment by JDGrimes** [comment]
> That's tricky, because it's a multi-line string, that is tokenized as single token and only line number it has is when it starts.

Nope, it isn't tokenized as a single token, each line is a separate token.

**Comment by aik099** [comment]
Then instead of removing `|| ($tokens[$nextComma]['line'] !== $valueLine)` you should be checking line of last token belonging to that array value.

**Comment by JDGrimes** [comment]
Yeah, but I can't figure out what that check is even for. I'm not sure it's needed.

**Comment by gsherwood** [comment]
Thanks for reporting. Issue is fixed now. Just needed to treat multi-line strings as one argument, like the sniff does for other multi-line values such as closures.

**Comment by JDGrimes** [comment]
Thanks for the quick fix @gsherwood! :tada:
